test/threads_exec.jl

```diff
@@ -729,6 +729,16 @@ let a = zeros(nthreads())
     @test a == [1:nthreads();]
 end
 
+# static schedule
+function _atthreads_static_schedule()
+    ids = zeros(Int, nthreads())
+    Threads.@threads static for i = 1:nthreads()
```

Comment:
This might be a stupid question: of which type is `static` in here? It’s not a symbol and looks like a keyword or a variable instead. As the tests are passing, I assume it’s not a typo.

I would like to propose that it should be a symbol instead, so that the scheduling may be controlled by a variable or similar (cf. `@spawnat :any ...` and `@spawnat 2 ...` from `Distributed`).

Reply:
It's used only at macro-expansion time so it doesn't correspond to any variables. We already have something like `@simd ivdep for ... end` so I think it's OK to have this interface.

But I think using `:static` actually makes sense.  It's handy to decide scheduler at run-time like

```julia
function f(...; scheduler=:static)
    @threads scheduler for ... end
end
```

Reply:
> It's handy to decide scheduler at run-time

This is actually a really fundamental decision though. Is the scheduler an object from an expression evaluated at run time, or is it known at macro expansion time, potentially allowing the macro to emit different code for different schedules?

Reply:
Do you think most (all?) of the schedulers need to do something during macro expansion?  Perhaps that's the reason why you are not in favor of `Threads.foreach` approach?

I was actually thinking that a (large) subset of schedulers can be set at run-time. We can throw a run-time error when a scheduler that has to be expanded by the macro is specified at runtime.

Reply:
Probably true, most schedules could be picked at run-time. I guess it's a good compromise to in general evaluate the expression, but recognize certain constant expressions like `:static` at macro expand time. I'll change it to that.